### PR TITLE
Replace custom `NodeSet` implementation by `@rdfjs/term-set`

### PR DESCRIPTION
**Pull request by martinmaillard**

### CODE REVIEW

File: src/node-set.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  toArray () {
+    return [...this]
+  }
+
+  toString () {
+    return `NodeSet(${this.size()}): [${this.toArray().join(', ')}]`
+  }
+}
+
+module.exports = NodeSet
```
Comment (bergos) [inline src/node-set.js]: Why not completely replace `NodeSet` with `term-set`? `[...]` would be even shorter than `.toArray`. The other methods are longer to inline them, so it depends on the usage.

Comment (martinmaillard) [reply src/node-set.js]: That's what I first started to do but the code got messy when I started replacing `addAll`. So I thought I would do it that way.

Any chance we can add something like `addAll` on `TermSet`?

Comment (martinmaillard) [reply src/node-set.js]: I removed `toString` and `toArray`. I'm going to keep `addAll` for now if you don't mind.

(may be replaced by https://github.com/rdfjs-base/term-set/pull/2 at some point?)

Comment (bergos) [reply src/node-set.js]: OK, I left a comment in the `TermSet` package. I think like you implemented it, it's the best option right now.

**Review by bergos — APPROVED** [review]

**Pull request merged**